src/middlewares/authentication-middleware.ts

```diff
@@ -1,22 +1,36 @@
-import { NextFunction, Request, Response } from 'express'
+import { Method, ZodiosPathsByMethod } from '@zodios/core'
+import { ZodiosRequestHandler } from '@zodios/express'
+import { Context } from '../config/zodios'
+import { Api } from '../config/zodios/api-schema'
+
+type Middleware = ZodiosRequestHandler<
+	Api,
+	Context,
+	Method,
+	ZodiosPathsByMethod<Api, Method>
+>
 
 export const verifyAuthentication = (...permissions: string[]) => {
-    return async (req: Request, res: Response, next: NextFunction) => {
-        if (!req.session) {
-        res.status(401).send('Not authenticated')
-            return
-        }
-    
-        if (permissions.length > 0) {
-            const { permissions: userPermissions } = req.session
-            const hasPermission = permissions.some(permission => userPermissions.includes(permission))
-        
-            if (!hasPermission) {
-                res.status(403).send('Not authorized')
-                return
-            }
-        }
-    
-        next()
-    }
-}    
\ No newline at end of file
+	const middleware: Middleware = (req, res, next) => {
+		if (!req.session) {
+			return res
+				.status(401)
+				.json({ error: 'Not authenticated', code: 401 })
```

Comment:
Here i format the error according to api definition else middleware would error saying it's not conforming to the api definition